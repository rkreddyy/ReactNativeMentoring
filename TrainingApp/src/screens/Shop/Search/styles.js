import { StyleSheet } from 'react-native';
import { BaseStyles } from '../../../app.styles';

export const SearchStyles = StyleSheet.create({
    wrapper: {
        backgroundColor: BaseStyles.colors.white,
        borderRadius: 10,
        borderColor: 'white',
        flexDirection: 'row',
    },
    input: {
        height: 40,
    },
    searchIcon: {
        marginLeft: BaseStyles.margin.l,
        marginRight: BaseStyles.margin.s,
        alignSelf: 'center',
    },
});
