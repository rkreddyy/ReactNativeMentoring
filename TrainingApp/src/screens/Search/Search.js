import React from 'react';
import { TextInput, View } from 'react-native';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';

import { SearchStyles } from './styles';
import { BaseStyles } from '../../app.styles';

const Search = () => {
    searchPlaceholder = 'Search for product'
    onSearch = (text) => {
        console.log(text);
    }
    return (
        <View style={SearchStyles.wrapper}>
            <FontAwesomeIcon
                style={SearchStyles.searchIcon}
                icon={faSearch}
                size={BaseStyles.fontSize.l}
                color={BaseStyles.colors.grey}
            />
            <TextInput
                onChangeText={text => onSearch(text)}
                style={SearchStyles.input}
                placeholder={'Search for product'}
                placeholderTextColor={BaseStyles.colors.gray}
            />
        </View>
    );
}

export default Search;
