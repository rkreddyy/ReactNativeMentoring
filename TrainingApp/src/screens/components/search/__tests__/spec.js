import React from 'react';
import { shallow } from 'enzyme';

import Search from '../search';

describe(`Search component`, () => {
    const renderComponent = props => shallow(<Search {...props} />);

    it('renders correctly', () => {
        const component = renderComponent({
            isSearchVisible: true
        });

        expect(component).toMatchSnapshot();
    });
});
