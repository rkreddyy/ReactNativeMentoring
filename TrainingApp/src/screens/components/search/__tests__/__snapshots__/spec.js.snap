// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00a8f3",
      "padding": 12,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "white",
        "borderRadius": 10,
        "flexDirection": "row",
      }
    }
  >
    <SearchIcon
      needToDisplay={true}
    />
    <TextInput
      allowFontScaling={true}
      placeholder="Search for product"
      rejectResponderTermination={true}
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
